Stripe/PublicHeaders/STPSourceParams.h

```diff
@@ -248,12 +248,32 @@ NS_ASSUME_NONNULL_BEGIN
  @param returnURL   The URL the customer should be redirected to after they have
  successfully verified the payment.
 
- @return an STPSourceParams object populated with the provided values
+ @return An STPSourceParams object populated with the provided values
  */
 + (STPSourceParams *)alipayParamsWithAmount:(NSUInteger)amount
                                    currency:(NSString *)currency
                                   returnURL:(NSString *)returnURL;
 
+
+/**
+ Creates params for a P24 source
+ @see https://stripe.com/docs/sources/p24#create-source
+
+ @param amount      The amount to charge the customer.
+ @param currency    The currency the payment is being created in (this must be 
+ EUR or PLN)
+ @param email       The email address of the account holder.
+ @param returnURL   The URL the customer should be redirected to after they have
+ @param name        The full name of the account holder (optional).
+
+ @return An STPSourceParams object populated with the provided values.
+ */
++ (STPSourceParams *)p24ParamsWithAmount:(NSUInteger)amount
+                                currency:(NSString *)currency
+                                   email:(NSString *)email
+                               returnURL:(NSString *)returnURL
+                                    name:(nullable NSString *)name;
```

Comment:
is there a pattern we're using for ordering these parameters?

Reply:
I don't think so? I've been putting all the optional ones at the end but otherwise I don't think there is a set pattern. Do you have a suggestion?

Reply:
I guess I would copy what we have in the [P24 Documentation](https://stripe.com/docs/sources/p24) which looks alphabetical after `type`:

```
type
amount
currency
(owner)email
(owner)name
(redirect)returnURL
```

Reply:
Hmm, that won't match the rest of the source params methods though? (which I think are basically alphabetical but with optional ones at the end?) Happy to change it going forward for all new apms starting here though.

Reply:
I think the important point is that `name` feels pretty separated from `email` now even though they're referring to the same "person".

Reply:
Changed.